addons/website/models/website.py

```diff
@@ -1342,3 +1342,99 @@ def _get_cached_values(self):
 
     def _get_cached(self, field):
         return self._get_cached_values()[field]
+
+    def _get_html_fields(self):
+        html_fields = [('ir_ui_view', 'arch_db')]
+        cr = self.env.cr
+        cr.execute(r"""
+            SELECT f.model, f.name
+                FROM ir_model_fields f
+                JOIN ir_model m ON m.id = f.model_id
+            WHERE f.ttype = 'html'
+                AND f.store = true
+                AND m.transient = false
+                AND f.model NOT LIKE 'ir.actions%'
+                AND f.model != 'mail.message'
+        """)
+        for model, name in cr.fetchall():
+            table = self.env[model]._table
+            if tools.table_exists(cr, table) and tools.column_exists(cr, table, name):
+                html_fields.append((table, name))
+        return html_fields
+
+    def _get_snippets_assets(self):
+        """Returns every snippet asset from the database in the form of a list :
+        [snippet_module, snippet_id, asset_version, asset_type, asset_id]
+        """
+        self.env.cr.execute(r"""
+            SELECT regexp_matches[1] AS snippet_module,
+                   regexp_matches[2] AS snippet_id,
+                   regexp_matches[3] AS asset_version,
+                   CASE WHEN regexp_matches[4]='scss' THEN 'css'
+                        ELSE regexp_matches[4]
+                   END AS asset_type,
+                   id AS asset_id
+            FROM
+                (SELECT REGEXP_MATCHES(PATH, '(\w*)\/.*\/snippets\/(\w*)\/(\d{3})\.(js|scss)'),
+                        id
+                FROM ir_asset
+                ORDER BY id ASC) AS regexp;
+        """)
+        return self.env.cr.fetchall()
+
+    def _is_snippet_used(self, snippet_module, snippet_id, asset_version, asset_type, html_fields):
+        snippet_occurences = []
+        # We check every snippet template definition
```

Comment:
I am pretty sure this isn't obvious for the next developer than will need to work on this code, if he didn't read the explanation
Maybe something a bit more elaborate would help IMHO.
```Check snippet template definition to avoid disabling its related assets. This special case is needed because snippet template definition do not have a `data-snippet` attribute (which is added during drag&drop).```

Indeed, I was wondering why it was needed and couldn't be done in the next loop, but the reason is there is no data-snippet

If you agree, you can add it, split it on 80 char length and rephrase it if you have better comment